ntbs-service/Services/NotificationChangesService.cs

```diff
@@ -0,0 +1,283 @@
+            }
+
+            // Clinical details page contains "site" models, which create their own audit entries
+            if (group.Any(log => log.EntityType == nameof(NotificationSite)))
+            {
+                // This could be done as part of other updates to the clinical details page..
+                var clinicalDetailsUpdate = group.Find(log => log.EntityType == nameof(ClinicalDetails));
+                if (clinicalDetailsUpdate != null)
+                    yield return clinicalDetailsUpdate;
+                // .. or, if sites are the only edited items, we want to "fake" a clinical details log entry
+                else
+                {
+                    var fakeClinicalDetailsLog = group.First().Clone();
+                    fakeClinicalDetailsLog.EntityType = nameof(ClinicalDetails);
+                    fakeClinicalDetailsLog.EventType = "Update";
+                    yield return fakeClinicalDetailsLog;
+                }
+
+                yield break;
+            }
+
+            // Social risks page contains sub-models, which create their own audit entries
+            var socialRisksUpdate = group.Find(log => log.EntityType == nameof(SocialRiskFactors));
+            if (socialRisksUpdate != null)
+            {
+                yield return socialRisksUpdate;
+                yield break;
+            }
+
+            // Travel and visitor details get saved together, but it actually makes sense to show them independently
+            if (group.Count == 2
+                && group.Any(log => log.EntityType == nameof(TravelDetails))
+                && group.Any(log => log.EntityType == nameof(VisitorDetails)))
+            {
+                foreach (var item in group)
+                {
+                    yield return item;
+                }
+
+                yield break;
+            }
+
+            // Transferring creates a flurry of related events.
+            // For each of these events we just want to have a single item in the history view 
+            var transferAlertLog = group.Find(log => log.EntityType == nameof(TransferAlert)
+                                                     || log.EntityType == nameof(TransferRejectedAlert));
+            if (transferAlertLog != null)
+            {
+                // - Request creates just the one:
+                //   - TransferAlert - Insert
+                if (transferAlertLog.EventType == "Insert")
+                {
+                    yield return transferAlertLog;
+                }
+                // - Acceptance creates:
+                //   - TreatmentEvent - Insert
+                //   - TreatmentEvent - Insert
+                //   - PreviousTbService - Insert
+                //   - HospitalDetails - Update
+                //   - TransferAlert - Update
+                else if (group.Any(log => log.EntityType == nameof(TreatmentEvent)))
+                {
+                    yield return transferAlertLog;
```

Comment:
Is this right? seems like it does the same thing as previous condition

Reply:
Yes, I split it off into a separate logic branch, as it will likely stop doing exactly the same thing once we need to build the details body of this event. But for the time being it happens to pick out the same event as the other branch :)